lib/charms/layer/openstack.py

```diff
@@ -95,13 +96,28 @@ def _save_creds(creds_data):
     else:
         attrs = creds_data
         endpoint = attrs['auth-url']
+
+    if 'ca-certificates' in creds_data:
+        # see K8s commit e3c8a0ceb66816433b095c4d734663e1b1e0e4ea
+        # K8s in-tree cloud provider code is not flexible enough
+        # to accept multiple certs that could be provided by Juju
+        # so we can grab the first one only and hope it is the
```

Comment:
If we end up going this route (see comment on juju-solutions/kubernetes#151), I don't think we should special-case things here because the current use-case can't support multiple credentials.  Rather, I'd prefer to have the charm and interface fully support the multiple creds and have the hack to randomly pick one done as close to the reason for it as possible, i.e. in the k8s charms.

Reply:
I see your point. I don't think this will ever happen in practice because generally endpoint certificates are all signed using a single CA.

Even the upstream OpenStack client written in python supports a single OS_CACERT option so I doubt the provider code will ever be different (especially considering that Cloud Controller Manager pods will soon replace this).

https://docs.openstack.org/python-openstackclient/queens/cli/man/openstack.html#options

A file pointed to by OS_CACERT could contain a chain of certificates but not multiple paths to PEM files.

https://tools.ietf.org/html/rfc8446#section-4.4.2
```
   certificate_list:  A sequence (chain) of CertificateEntry structures,
      each containing a single certificate and set of extensions
```

Making a list to be passed over a relation also potentially complicates the endpoint-tls-ca option - I would either have to special-case the fact that this config option results in a one-item list in the code or specify that a list should be provided instead for list config option somehow (CSV base64 values or a base64-encoded list in JSON doesn't seem like a good idea).

Anyway, I could change this but it seems to be an overkill. What do you think?